### PR TITLE
feature/volume_change_thesis

### PR DESCRIPTION
**Pull request by JohnVirtonomy**
Add test_3d_heart_volume_change adding volume change to heart simulation
- identify surface particles
- calculate surface area
- calculate ventricle volume changes
- export to csv-files

### CODE REVIEW

**Comment by JohnVirtonomy** [comment]
@Xiangyu-Hu for some reason the particles get destroyed after a few timsteps.
But my additions don't influence the particles. I ran the original test case (test_3d_heart_electromechanics) and it has the same issue (see image).
image

**Comment by Xiangyu-Hu** [comment]
> @Xiangyu-Hu for some reason the particles get destroyed after a few timsteps. But my additions don't influence the particles. I ran the original test case (test_3d_heart_electromechanics) and it has the same issue (see image). image

Do you say that the original case in master failed?

**Comment by JohnVirtonomy** [comment]
> Do you say that the original case in master failed?

Yes

**Review by BenceVirtonomy — APPROVED** [review]

File: SPHINXsys/src/shared/particle_dynamics/general_dynamics/general_interpolation.cpp [hunk 1]
```diff
@@ -49,7 +49,7 @@ namespace SPH
 			for (size_t n = 0; n != contact_neighborhood.current_size_; ++n)
 			{
 				Vecd normalized_weight_correction = B_ * weight_correction;
-				contact_neighborhood.W_ij_[n] -= contact_neighborhood.dW_ijV_j_[n] *
+				contact_neighborhood.W_ij_[n] -= contact_neighborhood.W_ij_[n] *
```
Comment (Xiangyu-Hu) [inline SPHINXsys/src/shared/particle_dynamics/general_dynamics/general_interpolation.cpp:52]: Don't use Wij, just divide Vj to cancel the effect.

**Comment by Xiangyu-Hu** [comment]
strange, the CI does not pass. Not able to build.

**Pull request merged**